Sources/Helpers/OAFavoritesHelper.mm

```diff
@@ -471,6 +479,52 @@ + (NSString *) checkEmoticons:(NSString *)text
     return [NSString stringWithString:tempString];
 }
 
++ (void) setupIcons
+{
+    NSString* path = [[NSBundle mainBundle] pathForResource:@"poi_categories" ofType:@"json"];
+    NSData *data = [NSData dataWithContentsOfFile:path];
+    NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:nil];
+
+    _poiIcons = [NSMutableDictionary new];
+    _flatPoiIcons = [NSMutableArray new];
+    
+    if (json)
+    {
+        NSDictionary *categories = json[@"categories"];
+        if (categories)
+        {
+            for (NSString *categoryName in categories.allKeys)
+            {
+                NSArray<NSString *> *icons = categories[categoryName][@"icons"];
+                if (icons)
+                {
+                    _poiIcons[categoryName] = icons;
+                    [_flatPoiIcons addObjectsFromArray:icons];
+                }
+            }
+        }
+    }
+}
+
++ (NSMutableDictionary<NSString *, NSArray<NSString *> *> *) getCategirizedIconNames
+{
+    if (!_poiIcons)
+        [self.class setupIcons];
```

Comment:
No need to add .class if calling a static method from the static method in the same class.